### ISSUE TITLE
Voxel size in elliptic transformed image

### ISSUE THREAD

**Issue by tischi** [BIOP/bigdataviewer-biop-tools#18]
@NicoKiaru 

We would like to assign voxel sizes to the elliptic transformed image. 
This is tricky as the size of every voxel in the transformed image can be different (in terms of what volume it represented in the original image). I guess thus we would actually need t a "voxel size image" (like a deformation field)?
(ping @bogovicj)

Do you have any ideas? I thought if we could compute the volume of a region in the transformed image in the original image this would get us something; but now I am confused, because even if we know this, how do we go from the volume to the voxel sizes along the three dimensions?

BTW: This is why I asked on the forum about how to compute the volume...
https://forum.image.sc/t/compute-volume-after-transformation/57728/2

**Comment by tischi** [comment]
Maybe (pseudo-code) one could create an n-dimensional distance field like this?

```
for d in dimensions:
   for v in voxels of an iterable region in the transformed image:
      pos = v.getPosition()
      neighborPos = v.move(d,1).getPosition()
      posInOriginalImage = transform.inverse().apply( pos )
      neighborPosInOriginalImage = transform.inverse().apply( neighborPos )

      distance_image[d, v.getPosition()] = distance( posInOriginalImage - neighborPosInOriginalImage )
```

If this is correct, from this one could then, btw, also compute the volume in the original image...

**Comment by tischi** [comment]
If above algorithm is correct, would it be OK for you if we (I can contribute the code) add this to the `Elliptic3DTransformerCommand` such that it would output not only the transformed image but in addition also three more sources with the corresponding local voxel dimensions in the original image? I think this could be very useful, because one could then still measure physical volumes and distances in the transformed image and also would have a visual representation of the deformations that have been applied to go from the original image to this one.

**Comment by tischi** [comment]
@bogovicj I have a feeling that [your code](https://github.com/imglib/imglib2-realtransform/blob/0d8b04b7d3a802a146e1d5b1ff47bfd79998f04b/src/main/java/net/imglib2/realtransform/inverse/RealTransformFiniteDerivatives.java#L48) is very similar to the algorithm that I proposed above? Would you say it essentially is the same idea?

**Comment by bogovicj** [comment]
@tischi , I agree, your pseudocode is doing something similar to that code of mine you pointed to.
